tests/test_board.py

```diff
@@ -3,85 +3,58 @@
-                [None, None, None, None, None, None, None, None, None, 'S'],
-                [None, None, None, None, 'C', None, None, None, None, None],
-                [None, 'D', None, None, 'C', None, None, None, None, None],
-                [None, 'D', None, None, 'C', None, None, None, None, None],
-                [None, None, None, None, None, None, None, None, None, None],
-                [None, None, None, None, None, None, 'B', 'B', 'B', 'B'],
-                [None, None, None, None, None, None, None, None, None, None],
-                [None, None, None, None, None, None, None, None, None, None],
-                ]
+       self.ships = Ships()
+       self.board_helper = BoardHelper(self.ships)
 
-       self.board_with_a_miss = [
-                ['AC', 'AC', 'AC', 'AC', 'AC', None, None, None, None, 'S'],
-                ['Miss', None, None, None, None, None, None, None, None, 'S'],
-                [None, None, None, None, None, None, None, None, None, 'S'],
-                [None, None, None, None, 'C', None, None, None, None, None],
-                [None, 'D', None, None, 'C', None, None, None, None, None],
-                [None, 'D', None, None, 'C', None, None, None, None, None],
-                [None, None, None, None, None, None, None, None, None, None],
-                [None, None, None, None, None, None, 'B', 'B', 'B', 'B'],
-                [None, None, None, None, None, None, None, None, None, None],
-                [None, None, None, None, None, None, None, None, None, None],
-                ]
-       
     def test_Hit_is_added_to_board_if_the_user_hit_a_ship(self):
         user_shot_choice = 'B1'
-        self.board.state = self.board_with_ships 
-        hit = True
-        self.maxDiff = None 
-        self.board.update(user_shot_choice, hit)
+        self.board.state = self.board_helper.generate_board_with_ships() 
+        shot_result = 'Hit'
+        self.board.update(user_shot_choice, shot_result)
 
-        self.assertEqual(self.board.state, self.board_with_a_hit)
+        self.assertEqual(self.board.state, self.board_helper.generate_board_with_hit())
 
     def test_Miss_is_added_to_board_if_the_user_Misses_a_ship(self):
         user_shot_choice = 'A2'
-        self.board.state = self.board_with_ships 
-        hit = False
-        self.maxDiff = None 
-        self.board.update(user_shot_choice, hit)
+        self.board.state = self.board_helper.generate_board_with_ships() 
+        shot_result = 'Miss'
+        self.board.update(user_shot_choice, shot_result)
 
-        self.assertEqual(self.board.state, self.board_with_a_miss)
+        self.assertEqual(self.board.state, self.board_helper.generate_board_with_a_miss())
+    
+    def test_when_ship_sinks_all_positions_where_ship_was_are_updated_to_str_Sunk(self):
+        user_shot_choice = 'A1'
+        self.board.state = self.board_helper.generate_board_with_ships() 
+        self.board.state[0][0] = {
```

Comment:
ooof this seems like a pain for every time you want to place a ship on a board. the `[0][0]` seems a little opaque to me.